network/basics/uri.py

```diff
@@ -367,14 +374,24 @@ def main():
     status_code = [int(x) for x in list(module.params['status_code'])]
     socket_timeout = module.params['timeout']
 
-    # Grab all the http headers. Need this hack since passing multi-values is currently a bit ugly. (e.g. headers='{"Content-Type":"application/json"}')
     dict_headers = {}
+
+    # If body_format is json or yaml, encore the body (wich can be a dict or a list) and automatically sets the Content-Type header
+    if body_format == 'json':
+        body = json.dumps(body)
+        dict_headers['Content-Type'] = 'application/json'
+    elif body_format == 'yaml':
+        body = yaml.dump(body)
```

Comment:
The YAML module is never imported anywhere, additionally the YAML python module is not a required module for a ansible managed machine.

Does this provide much benefit?  I have never seen an API accept `application/yaml` anyway.